crates/bevy_render/src/view/mod.rs

```diff
@@ -116,78 +117,107 @@ pub struct ViewUniformOffset {
     pub offset: u32,
 }
 
-#[derive(Clone)]
-pub enum ViewMainTexture {
-    Hdr {
-        hdr_texture: TextureView,
-        sampled_hdr_texture: Option<TextureView>,
-
-        ldr_texture: TextureView,
-    },
-    Sdr {
-        texture: TextureView,
-        sampled_texture: Option<TextureView>,
-    },
-}
-
-impl ViewMainTexture {
-    pub fn texture(&self) -> &TextureView {
-        match self {
-            ViewMainTexture::Hdr { hdr_texture, .. } => hdr_texture,
-            ViewMainTexture::Sdr { texture, .. } => texture,
-        }
-    }
-}
-
 #[derive(Component)]
 pub struct ViewTarget {
-    pub main_texture: ViewMainTexture,
-    pub out_texture: TextureView,
-    pub out_texture_format: TextureFormat,
+    main_textures: MainTargetTextures,
+    main_texture_format: TextureFormat,
+    /// 0 represents `texture_a`, 1 represents `texture_b`
+    main_texture: AtomicUsize,
```

Comment:
should we force post processing to use &mut ViewTarget? i'm not sure but it feels like the atomic here could mask issues with concurrent writers, where there's no guarantee the render passes would be in the same order as the post_process_write requests.

Reply:
Yeah this is currently valid. I would like to revisit what mutability might look like here though. There _is_ world where we choose to make node processing parallel. Thats one of the reasons we're restricting to `&World` in the first place (to make graph execution read-only).

I know a number of people (including myself) have explored the idea to move this stuff into systems. Still not sure how I feel about it, but it _would_ allow more granular read/writes in a parallel context (force linear command encoding for post processing effects, by nature of the mutable access to ViewTarget, allow parallel command encoding for most other things, as long as the systems are read only).